src/HTTP/Factory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/* Copyright (c) 2016 Fabian Schmid <fs@studer-raimann.ch> Extended GPL, see docs/LICENSE */
+namespace ILIAS\HTTP;
+
+use ILIAS\HTTP\Cookies\CookieJar;
+use ILIAS\HTTP\Response\rendering\RenderingException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Interface Factory
```

Comment:
@d3r1w Concerning the performance of clone vs constructor. The gain seems not to be huge. @shecken wrote above:

> You're right. Looks like clone is a little cheaper then new. I don't know how often it will be called in this case, but i think it could be negligible. Benchmark runs i found shows differences about 0,2 seconds if cloning 4 object in 100.000 cycles.

I would personally prefer clone. But it seems to be more about personal preference in this case than performance. However, if you write code that will clone a lot (such as in the UI-Framework), definitely use clone instead of new.

However, we should not forget, that 400'000 sound like a lot, but might be quickly reached in an application like ILIAS AND 0.2s is definitely a huge performance loss.